File: Python/백준/Gold/1669. 멍멍이 쓰다듬기/멍멍이 쓰다듬기.py
# 1669 멍멍이 쓰다듬기
import sys
input = sys.stdin.readline

X, Y = map(int, input().split())

target= Y-X

if target in (0, 1, 2, 3):
    print(target)
else:
    day = int(target ** (1/2) )
    if target == day ** 2:
        print(day+day-1)
    elif day * (day+1) >= target :
        print(day+day)
    # elif (day + 1) ** 2 >= target:
    else:
        print(day+day+1)

File: node.js/백준/Gold/1669. 멍멍이 쓰다듬기/멍멍이 쓰다듬기.js
// 1669 멍멍이 쓰다듬기
const fs = require("fs");
const [x, y] = fs.readFileSync('/dev/stdin').toString().split(" ").map(Number);

const target = y-x;

if (target in [0, 1, 2, 3]) {
    console.log(target);
} else {
    const day = parseInt(Math.sqrt(target));
    if (target === day*day) {
        console.log(day*2-1);
    } else if (target <= day*(day+1)){
        console.log(day*2);
    } else {
        console.log(day*2+1);
    }
}